lib/screens/registration_screen.dart

```diff
@@ -0,0 +1,179 @@
+  final formkey = GlobalKey<FormState>();
+
+  final TextEditingController email = TextEditingController();
+  final TextEditingController password = TextEditingController();
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        centerTitle: true,
+        title: const Text('Register Page'),
+      ),
+      body: Padding(
+        padding: const EdgeInsets.all(15.0),
+        child: Form(
+          key: formkey,
+          child: Column(
+            children: [
+              TextFormField(
+                controller: email,
+                enableSuggestions: false,
+                keyboardType: TextInputType.emailAddress,
+                autofillHints: const [AutofillHints.email],
+                validator: (email) {
+                  if (email!.isEmpty) {
+                    return '* Please enter email here';
+                  } else if (!EmailValidator.validate(email)) {
+                    return '* Enter a Valid Email Address';
+                  } else {
+                    return null;
+                  }
+                },
+                decoration: const InputDecoration(
+                    labelText: 'Email Address', border: OutlineInputBorder()),
+              ),
+              const SizedBox(
+                height: 15,
+              ),
+              TextFormField(
+                controller: password,
+                obscureText: true,
+                enableSuggestions: false,
+                validator: (value) {
+                  if (value!.isEmpty) {
+                    return '* Please enter password here';
+                  } else if (value.length < 6) {
+                    return 'Password should contains atleast 6 letters';
+                  } else {
+                    return null;
+                  }
+                },
+                decoration: const InputDecoration(
+                    labelText: 'Password', border: OutlineInputBorder()),
+              ),
+              Center(
+                child: ElevatedButton(
+                    onPressed: () async {
+                      final form = formkey.currentState!;
+                      if (form.validate()) {
+                        final email_ = email.text;
+                        final password_ = password.text;
+                      }
+
+                      await Firebase.initializeApp(
+                        options: DefaultFirebaseOptions.currentPlatform,
+                      );
+                      final email_ = email.text;
+                      final password_ = password.text;
+                      try {
+                        final UserCredential = await FirebaseAuth.instance
+                            .createUserWithEmailAndPassword(
+                                email: email_, password: password_);
+
+                        print(UserCredential);
+                      } on FirebaseAuthException catch (e) {
```

Comment:
make a repo file in which all the firebase business code are written and here just call the method from there